hal/emac_api.h

```diff
@@ -134,6 +138,13 @@ typedef void (*emac_set_link_input_cb_fn)(emac_interface_t *emac, emac_link_inpu
  */
 typedef void (*emac_set_link_state_cb_fn)(emac_interface_t *emac, emac_link_state_change_fn state_cb, void *data);
 
+/** Add device to a multicast group
+ *
+ * @param emac     Emac interface
+ * @param address  An multicast group IPv4 address
+ */
+typedef void (*emac_add_multicast_group)(emac_interface_t *emac, uint8_t *address);
```

Comment:
Is this taking in a string representation of an IP address or a byte representation? Should it actually be taking in the nsapi_addr_t struct?